.gitlab/includes/hip_pipeline.yml

```
@@ -0,0 +1,78 @@
+# Copyright (c) 2023 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file BOOST_LICENSE_1_0.rst or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+include:
+  - local: '.gitlab/includes/common_pipeline.yml'
+  - local: '.gitlab/includes/common_spack_pipeline.yml'
+
+.variables_gcc12_hip5_config:
+  variables:
+    ARCH: linux-ubuntu22.04-zen3
+    BUILD_TYPE: Debug
+    COMPILER: gcc@12.2.0
+    CXXSTD: 20
+    GPU_TARGET: 'gfx90a'
+    SPACK_SPEC: "pika@main arch=$ARCH %${COMPILER} +rocm +stdexec amdgpu_target=${GPU_TARGET} \
+                 malloc=system cxxstd=$CXXSTD ^boost@1.79.0 ^hwloc@2.6.0 ^hip@5.6 ^llvm~gold \
+                 ^fmt@10.0.0 ^whip amdgpu_target=${GPU_TARGET} \
```

Not necessary since we enforce them to be the same:
```suggestion
                 ^fmt@10.0.0 ^whip \
```